compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -630,7 +630,14 @@ public BLangNode transform(UnionTypeDescriptorNode unionTypeDescriptorNode) {
         BLangFiniteTypeNode bLangFiniteTypeNode = (BLangFiniteTypeNode) TreeBuilder.createFiniteTypeNode();
         for (TypeDescriptorNode finiteTypeEl : finiteTypeElements) {
             SingletonTypeDescriptorNode singletonTypeNode = (SingletonTypeDescriptorNode) finiteTypeEl;
-            BLangLiteral literal = createSimpleLiteral(singletonTypeNode.simpleContExprNode());
+            BLangLiteral literal;
+            Node simpleContExprNode = singletonTypeNode.simpleContExprNode();
+            if (simpleContExprNode.kind() == SyntaxKind.UNARY_EXPRESSION) {
+                UnaryExpressionNode unaryExpr = (UnaryExpressionNode) simpleContExprNode;
+                literal = createSimpleLiteral(unaryExpr.expression(), unaryExpr.unaryOperator().kind(), true);
+            } else {
+                literal = createSimpleLiteral(simpleContExprNode, SyntaxKind.NONE, true);
```

Comment:
I think this is already handles inside `createSimpleLiteral()` method?

Reply:
removed